spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/actions/DeleteOrphanFilesSparkAction.java

```diff
@@ -326,6 +344,96 @@ private static FlatMapFunction<Iterator<String>, String> listDirsRecursively(
     };
   }
 
+  @VisibleForTesting
+  static List<String> findOrphanFiles(SparkSession spark,
+                                      Dataset<Row> actualFileDF,
+                                      Dataset<Row> validFileDF,
+                                      Map<String, String> equalSchemes,
+                                      Map<String, String> equalAuthorities,
+                                      PrefixMismatchMode prefixMismatchMode) {
+    Map<String, String> equalSchemesMap = flattenMap(equalSchemes);
+    Map<String, String> equalAuthoritiesMap = flattenMap(equalAuthorities);
+
+    Dataset<PathProxy> normalizedActualFileDF = actualFileDF.mapPartitions(
```

Comment:
I think I liked `FileMetadata` more. I'd also consider changing vars a bit and dropping `as` calls.

```
Dataset<FileMetadata> actualFileMetadataDS = actualFileDF.mapPartitions(
    toFileMetadata(equalSchemes, equalAuthorities),
    Encoders.bean(FileMetadata.class));

Dataset<FileMetadata> validFileMetadataDS = validFileDF.mapPartitions(
    toFileMetadata(equalSchemes, equalAuthorities),
    Encoders.bean(FileMetadata.class));
```

Reply:
Reverted back to `FileMetadata` and removed `as`